usr.sbin/bhyve/gdb.c

```diff
@@ -187,7 +210,85 @@ static const int gdb_regsize[] = {
 	4,
 	4
 };
+#endif
+
+#ifdef __aarch64__
+static const int gdb_regset[] = {
+	VM_REG_GUEST_X0,
+	VM_REG_GUEST_X1,
+	VM_REG_GUEST_X2,
+	VM_REG_GUEST_X3,
+	VM_REG_GUEST_X4,
+	VM_REG_GUEST_X5,
+	VM_REG_GUEST_X6,
+	VM_REG_GUEST_X7,
+	VM_REG_GUEST_X8,
+	VM_REG_GUEST_X9,
+	VM_REG_GUEST_X10,
+	VM_REG_GUEST_X11,
+	VM_REG_GUEST_X12,
+	VM_REG_GUEST_X13,
+	VM_REG_GUEST_X14,
+	VM_REG_GUEST_X15,
+	VM_REG_GUEST_X16,
+	VM_REG_GUEST_X17,
+	VM_REG_GUEST_X18,
+	VM_REG_GUEST_X19,
+	VM_REG_GUEST_X20,
+	VM_REG_GUEST_X21,
+	VM_REG_GUEST_X22,
+	VM_REG_GUEST_X23,
+	VM_REG_GUEST_X24,
+	VM_REG_GUEST_X25,
+	VM_REG_GUEST_X26,
+	VM_REG_GUEST_X27,
+	VM_REG_GUEST_X28,
+	VM_REG_GUEST_X29,
+	VM_REG_GUEST_LR,
+	VM_REG_GUEST_SP,
+	VM_REG_GUEST_PC,
+	VM_REG_GUEST_CPSR,
+};
+
+static const int gdb_regsize[] = {
+	8,
```

Comment:
Maybe consider adding designators `[0] = 8]` for readability? If you're willing to use a gnu extension (which clang supports) you can even make this super compact:
```
const int gdb_regsize[] = {
	[0 ... 32] = 8,
	4,
};
```

Reply:
I also wonder if there should be a _Static_assert that the two arrays are the same size?

Reply:
I added the _Static_assert, but indeed will avoid changing the amd64 code for now. Those arrays will be mostly obsolete once support for xmlRegisters is done.